autopush/web/webpush.py

```diff
@@ -431,7 +431,7 @@ def validate_auth(self, d):
                                  headers={"www-authenticate": PREF_SCHEME})
         if jwt['aud'] != "{}://{}".format(
                 self.context["conf"].endpoint_scheme or "http",
```

Comment:
Oh `endpoint_url` might be more correct here vs endpoint_scheme + hostname. Assuming `aud` would naturally come from the subscription info created from `make_endpoint` (which uses `endpoint_url`). endpoint_url also includes the port (on non standard ports)

Reply:
yeah, that's the trickier part. It doesn't. "aud" comes from whoever is creating the VAPID signature, so it should match whatever the publicly visible push endpoint would be, non-standard port and all. It also should only be the scheme, host (and non-standard port) so any additional path stuff needs to be dropped.

Reply:
Sure it comes from whoever but they would *likely* look there e.g. https://developers.google.com/web/updates/2016/07/web-push-interop-wins#payload and `endpoint_url` seem to be exactly this (I think).